### PR TITLE
append special characters to reg-ex password validation for front-end…

### PR DESCRIPTION
**Pull request by fipa98**
… and back-end

### CODE REVIEW

**Comment by eliasvagan** [comment]
What story is this attached to?

File: backend/Extensions/NewUserExtensions.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static class NewUserExtensions
 
         public static bool IsValidPassword(this NewUserResource newUser)
         {
-            const string passwordRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\\$%\\^&\\*])(?=.{8,})";
+            const string passwordRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#-_\\$%\\^&\\*\\?\\€\\,\\;\\.\\:])(?=.{8,})";
```
Comment (MittBrukernavn) [inline backend/Extensions/NewUserExtensions.cs]: the character `-` has a special meaning when it's within brackets. Having `#-_` will allow any character with an ASCII value between 35 and 95.
```suggestion
            const string passwordRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\\-_\\$%\\^&\\*\\?\\€\\,\\;\\.\\:])(?=.{8,})";
```

File: frontend/src/pages/Register/index.jsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ const Register = () => {
   };
 
   const validPasswordFormat = (_password) => {
-    const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/; // eslint-disable-line no-useless-escape
+    const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#-_\$%\^&\*\?\€\,\;\.\:])(?=.{8,})/; // eslint-disable-line no-useless-escape
```
Comment (MittBrukernavn) [inline frontend/src/pages/Register/index.jsx]: Same as above: `-` should be escaped
```suggestion
    const re = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\-_\$%\^&\*\?\€\,\;\.\:])(?=.{8,})/; // eslint-disable-line no-useless-escape
```

**Comment by fipa98** [comment]
@eliasvagan This is a bug-type issue, not part of any story
@MittBrukernavn Will fix that
@Letaguator Lets look at that tomorrow

**Pull request merged**